leku/src/main/java/com/adevinta/leku/geocoder/places/GooglePlacesDataSource.kt

```diff
@@ -21,30 +20,27 @@ private const val PLACE_BY_ID_WAITING_TIME: Long = 3
 
 class GooglePlacesDataSource(private val geoDataClient: PlacesClient) {
 
-    fun getFromLocationName(query: String, latLngBounds: LatLngBounds): Single<List<Address>> {
+    fun getFromLocationName(query: String, latLngBounds: LatLngBounds): List<Address> {
         val locationBias = RectangularBounds.newInstance(
             latLngBounds.southwest,
             latLngBounds.northeast)
-        return Single.defer {
-            val findAutocompletePredictionsRequest = FindAutocompletePredictionsRequest
-                .builder()
-                .setQuery(query)
-                .setLocationBias(locationBias)
-                .build()
-            val results = geoDataClient.findAutocompletePredictions(findAutocompletePredictionsRequest)
-            try {
-                Tasks.await(results, PREDICTIONS_WAITING_TIME, TimeUnit.SECONDS)
-            } catch (ignored: ExecutionException) {
-            } catch (ignored: InterruptedException) {
-            } catch (ignored: TimeoutException) {
-            }
+        val findAutocompletePredictionsRequest = FindAutocompletePredictionsRequest
+            .builder()
+            .setQuery(query)
+            .setLocationBias(locationBias)
+            .build()
+        val results = geoDataClient.findAutocompletePredictions(findAutocompletePredictionsRequest)
+        try {
+            Tasks.await(results, PREDICTIONS_WAITING_TIME, TimeUnit.SECONDS)
```

Comment:
What are you trying to achieve here?

Reply:
Here I only removed the Single.defer wrap. If I remember well this was a copy&paste from Google documentation.
🤔  Probably this must be reviewed later on.